src/plot.cpp

```diff
@@ -980,16 +1019,552 @@ RGBColor random_color(void)
+   * surface edges oriented horizontally in the rendering.
+   *
+   * Note that a vector of vectors is required rather than a 2-tensor,
+   * since the stack size varies within each column.
+   */
+#ifdef _OPENMP
+  const int n_threads = omp_get_max_threads();
+#else
+  const int n_threads = 1;
+#endif
+  std::vector<std::vector<std::vector<TrackSegment>>> this_line_segments(
+    n_threads);
+  for (int t = 0; t < n_threads; ++t) {
+    this_line_segments[t].resize(pixels_[0]);
+  }
+
+  // The last thread writes to this, and the first thread reads from it.
+  std::vector<std::vector<TrackSegment>> old_segments(pixels_[0]);
+
+#pragma omp parallel
+  {
+
+#ifdef _OPENMP
+    const int n_threads = omp_get_max_threads();
+    const int tid = omp_get_thread_num();
+#else
+    int n_threads = 1;
+    int tid = 0;
+#endif
+
+    SourceSite s; // Where particle starts from (camera)
+    s.E = 1;
+    s.wgt = 1;
+    s.delayed_group = 0;
+    s.particle = ParticleType::photon; // just has to be something reasonable
+    s.parent_id = 1;
+    s.progeny_id = 2;
+    s.r = camera_position_;
+
+    Particle p;
+    s.u.x = 1.0;
+    s.u.y = 0.0;
+    s.u.z = 0.0;
+    p.from_source(&s);
+
+    int vert = tid;
+    for (int iter = 0; iter <= pixels_[1] / n_threads; iter++) {
+
+      // Save bottom line of current work chunk to compare against later
+      // I used to have this inside the below if block, but it causes a
+      // spurious line to be drawn at the bottom of the image. Not sure
+      // why, but moving it here fixes things.
+      if (tid == n_threads - 1)
+        old_segments = this_line_segments[n_threads - 1];
+
+      if (vert < pixels_[1]) {
+
+        for (int horiz = 0; horiz < pixels_[0]; ++horiz) {
+
+          // Generate the starting position/direction of the ray
+          if (orthographic_width_ == 0.0) { // perspective projection
+            double this_phi = -horiz_fov_radians / 2.0 + dphi * horiz;
+            double this_mu = -vert_fov_radians / 2.0 + dmu * vert + M_PI / 2.0;
+            Direction camera_local_vec;
+            camera_local_vec.x = std::cos(this_phi) * std::sin(this_mu);
+            camera_local_vec.y = std::sin(this_phi) * std::sin(this_mu);
+            camera_local_vec.z = std::cos(this_mu);
+            s.u = camera_local_vec.rotate(camera_to_model);
+          } else { // orthographic projection
+            s.u = looking_direction;
+
+            double x_pix_coord = (static_cast<double>(horiz) - p0 / 2.0) / p0;
+            double y_pix_coord = (static_cast<double>(vert) - p1 / 2.0) / p0;
+            s.r = camera_position_ +
+                  cam_yaxis * x_pix_coord * orthographic_width_ +
+                  cam_zaxis * y_pix_coord * orthographic_width_;
+          }
+
+          p.from_source(&s); // put particle at camera
+          bool hitsomething = false;
+          bool intersection_found = true;
+          int loop_counter = 0;
+          const int max_intersections = 1000000;
```

Comment:
There are a couple of constants in here that are a little hidden. What would you say to making them members of the class?

Reply:
I only see max intersections as a constant. What others are you talking about? The others are completely arbitrary and could be set to any value. I think.

Reply:
anyway, i did this for max_intersections!